### PR TITLE
implemented autocomplete in search query

### PR DESCRIPTION
**Pull request by mrubakh**
- implemented autocomplete in the Items#index view
- Installed EasyAutoComplete gem and implemented dynamic AJAX requests that took the list of item names
- fixed cucumber tests

### CODE REVIEW

File: app/assets/javascripts/messages.coffee [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# jQuery(document).on 'turbolinks:load', ->
```
Comment (amycsong98) [inline app/assets/javascripts/messages.coffee:1]: does this file need to be present?

**Comment by amycsong98** [comment]
can you add a picture of what it looks like

**Comment by mrubakh** [comment]
Screen Shot 2020-12-09 at 1 39 25 PM

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -98,14 +95,15 @@ def create
   end
   
   def search
-    if !params[:search].empty?
+    if params[:search] == nil
```
Comment (amycsong98) [inline app/controllers/items_controller.rb:98]: if `params[:search].nil?`

Comment (amycsong98) [reply app/controllers/items_controller.rb:98]: wait why was the logic changed it from not being empty to it being null

File: app/assets/javascripts/messages.coffee (hunk 1 above)
Comment (mrubakh) [reply app/assets/javascripts/messages.coffee:1]: whoops yeah this shouldnt be here

File: app/controllers/items_controller.rb (hunk 2 above)
Comment (amycsong98) [reply app/controllers/items_controller.rb:98]: ah nvm i get it

File: app/models/item.rb [hunk 3]
```diff
@@ -9,7 +9,7 @@ class Item < ApplicationRecord
     validates :price, :presence => true, format: { with: /\A\d+(?:\.\d{0,2})?\z/ }, numericality: { greater_than: 0, less_than: 1000000 }
     validates :description, :presence => true, length: {maximum: 100}
     validate :image_type
-    #size_range: 1..3.megabytes
```
Comment (amycsong98) [inline app/models/item.rb:12]: why was this line deleted?

File: app/views/items/index.html.erb [hunk 4]
```diff
@@ -76,4 +67,14 @@
         <% end %>
       </tbody>
     </table>
+    <%= form_tag(item_search_path, :method => "get") do %>
```
Comment (amycsong98) [inline app/views/items/index.html.erb:70]: does changing this break tests?

File: config/environments/development.rb [hunk 5]
```diff
@@ -52,6 +52,8 @@
   # number of complex assets.
   config.assets.debug = true
   config.assets.compile = true
+  
+  config.web_console.whitelisted_ips = '69.204.8.105'
```
Comment (amycsong98) [inline config/environments/development.rb:56]: oof should this go in config?

File: app/assets/javascripts/application.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+//= require underscore
```
Comment (Allegraknox) [inline app/assets/javascripts/application.js:1]: Need to include this file?

File: config/routes.rb [hunk 7]
```diff
@@ -6,7 +6,7 @@
     resources :users, :only=>[:show, :edit, :update]
     root to: "items#index"
     
-    get '/search' => 'items#search', :as => 'search_item'
+    get 'item_search', to: 'items#search'
```
Comment (amycsong98) [inline config/routes.rb:9]: does changing this break tests?

File: app/assets/javascripts/messages.coffee (hunk 1 above)
Comment (Allegraknox) [inline app/assets/javascripts/messages.coffee:1]: Is this whole file commented out? Need to include it at all?

File: app/models/item.rb (hunk 3 above)
Comment (mrubakh) [reply app/models/item.rb:12]: It was a comment left in here a long time ago - we have no logic to check size of the image anyways

File: app/views/items/index.html.erb (hunk 4 above)
Comment (mrubakh) [reply app/views/items/index.html.erb:70]: what do you mean?

File: public/packs/manifest.json [hunk 8]
```diff
@@ -1,14 +1,24 @@
 {
-  "application.js": "/packs/js/application-9afcbb5693aa87623e69.js",
-  "application.js.map": "/packs/js/application-9afcbb5693aa87623e69.js.map",
+  "application.js": "/packs/js/application-f3151c7bc3b441a5f9bf.js",
```
Comment (amycsong98) [inline public/packs/manifest.json:2]: 👀

File: config/environments/development.rb (hunk 5 above)
Comment (mrubakh) [reply config/environments/development.rb:56]: This was necessary for me and it doesn't hurt? I guess I can omit this as well.

File: app/controllers/items_controller.rb [hunk 9]
```diff
@@ -98,14 +95,15 @@ def create
...
+      q = params[:q].downcase
+      @item = Item.search(q)
+      render json: @item.select(:name)
+    else
```
Comment (Allegraknox) [inline app/controllers/items_controller.rb:102]: Did u cover this in tests (if it broke anything), i see you changed one test line but dont see anything else

File: app/assets/javascripts/application.js (hunk 6 above)
Comment (mrubakh) [reply app/assets/javascripts/application.js:1]: yep, see https://apidock.com/rails/String/underscore

File: config/routes.rb (hunk 7 above)
Comment (mrubakh) [reply config/routes.rb:9]: all tests still pass

File: app/assets/javascripts/messages.coffee (hunk 1 above)
Comment (mrubakh) [reply app/assets/javascripts/messages.coffee:1]: yep I will omit this file, good point.

File: config/routes.rb (hunk 7 above)
Comment (amycsong98) [reply config/routes.rb:9]: 100% coverage?

File: app/assets/stylesheets/custom.css.scss [hunk 10]
```diff
@@ -99,19 +108,30 @@ p {
...
 }
 
-
 footer {
-	position: fixed;
-	bottom: 0;
-	width: 100%;
-	padding: 1rem;
-	text-align: center;
+	margin-top: 45px;
```
Comment (mariamgrig) [inline app/assets/stylesheets/custom.css.scss:112]: Again, this is the previous version of css before I pushed, but that's fine, I will just merge again after we merge this

File: public/packs/manifest.json (hunk 8 above)
Comment (mrubakh) [reply public/packs/manifest.json:2]: I can probably also omit this file - just wanted to be sure that it wouldnt cause anything to break

File: app/views/items/index.html.erb (hunk 4 above)
Comment (amycsong98) [reply app/views/items/index.html.erb:70]: just was checking it didnt break any tests

File: app/assets/javascripts/application.js (hunk 6 above)
Comment (mariamgrig) [reply app/assets/javascripts/application.js:1]: Yeah I am not sure if we need this either

File: config/environments/development.rb (hunk 5 above)
Comment (amycsong98) [reply config/environments/development.rb:56]: im not sure if it's sens- i t-ve data or not

File: app/assets/stylesheets/custom.css.scss (hunk 10 above)
Comment (amycsong98) [reply app/assets/stylesheets/custom.css.scss:112]: `git pull origin main` @mrubakh

File: app/controllers/items_controller.rb (hunk 9 above)
Comment (mrubakh) [reply app/controllers/items_controller.rb:102]: Yep, I made sure all tests work

File: app/assets/stylesheets/custom.css.scss (hunk 10 above)
Comment (mrubakh) [reply app/assets/stylesheets/custom.css.scss:112]: I totally did that... git is so annoying

File: config/routes.rb (hunk 7 above)
Comment (mrubakh) [reply config/routes.rb:9]: nope - I haven't done tests yet
